src/Model/TransportModel/tsp-adv.f90

```diff
@@ -127,6 +130,71 @@ subroutine adv_ar(this, dis, ibound)
     return
   end subroutine adv_ar
 
+  !> @brief  Calculate maximum time step length
+  !!
+  !!  Return the largest time step that meets stability constraints
+  !<
+  subroutine adv_dt(this, dtmax, msg, thetam)
+    ! dummy
+    class(TspAdvType) :: this !< this instance
+    real(DP), intent(out) :: dtmax !< maximum allowable dt subject to stability constraint
+    character(len=*), intent(inout) :: msg !< package/cell dt constraint message
+    real(DP), dimension(:), intent(in) :: thetam !< porosity
+    ! local
+    integer(I4B) :: n
+    integer(I4B) :: m
+    integer(I4B) :: ipos
+    integer(I4B) :: nrmax
+    character(len=LINELENGTH) :: cellstr
+    real(DP) :: dt
+    real(DP) :: qmax
+    real(DP) :: qnmsumpos
+    real(DP) :: qnmsumneg
+    real(DP) :: qnm
+    real(DP) :: cell_volume
+    dtmax = DNODATA
+    nrmax = 0
+    msg = ''
+
+    ! If ats_percel not specified by user, then return without making
+    ! the courant time step calculation
+    if (this%ats_percel == DNODATA) then
+      return
+    end if
+
+    ! Calculate time step lengths based on stability constraint for each cell
+    ! and store the smallest one
+    do n = 1, this%dis%nodes
+      if (this%ibound(n) == 0) cycle
+      qnmsumneg = DZERO
+      qnmsumpos = DZERO
+      do ipos = this%dis%con%ia(n) + 1, this%dis%con%ia(n + 1) - 1
+        if (this%dis%con%mask(ipos) == 0) cycle
+        m = this%dis%con%ja(ipos)
+        if (this%ibound(m) == 0) cycle
```

Comment:
You are right about this.  The variable IBOUND has been around for 40 years and many of us know what it means, but this could be better named and handled, but that should probably happen in a separate and dedicated PR.  FYI, negative IBOUND values, values of zero, and values greater than zero mean that the cell is a constant head/conc cell, an inactive cell, or an active cell, respectively.